sfall/LoadGameHook.cpp

```diff
@@ -276,6 +277,45 @@ static void __declspec(naked) LoadGame() {
 		retn;
 	}
 }
+
+void ReadExtraGameMsgFiles()
+{
+	int read;
+	std::string names;
+
+	names.resize(256);
+
+	while ((read = GetPrivateProfileStringA("Misc", "ExtraGameMsgFileList", "",
+		(LPSTR)names.data(), names.size(), ".\\ddraw.ini")) == names.size() - 1)
+		names.resize(names.size() + 256);
+
+	if (names.empty())
+		return;
+
+	names.resize(names.find_first_of('\0'));
+	names.append(",");
+
+	int begin = 0;
+	int end;
+	int length;
+
+	while ((end = names.find_first_of(',', begin)) != std::string::npos)
+	{
+		length = end - begin;
+
+		if (length > 0)
+		{
+			std::string path = "game\\" + names.substr(begin, length) + ".msg";
```

Comment:
I think you should be able to debug a C++ dll with external executable on an Express version of Visual Studio. All you have to do is supply a path to the executable in the General Properties -> Debugging -> Command section of solution properties. I also believe that the compiled dll has to be in the same directory as the executable in this case. The next step is to simply start a debugging session.

You handled the initialization just fine. Too bad it doesn't help, but it's beneficial to do it nonetheless.

Here's the [link to the dll](https://dl.dropboxusercontent.com/u/29728588/ddraw.dll) compiled with VS Community 2015.